src/controllers/colorjsproxy.cpp

```diff
@@ -1,6 +1,24 @@
+#include <cstdint>
+#include <cmath>
 #include "controllers/colorjsproxy.h"
 #include "preferences/hotcuecolorpalettesettings.h"
 
+namespace {
+    double colorDistance(QColor a, QColor b)
+    {
+      long mean_red = ((long)a.red() + (long)b.red()) / 2;
```

Comment:
can you explain this approach further or link the article/paper where you got this from?

Reply:
Sure, it's the low-cost approximation from this article:
https://www.compuphase.com/cmetric.htm

Reply:
Ok that makes sense. I'd put a reference to that in the code.

Reply:
You're right, I'll add a comment.